tests/datasets/plans/translator/test_mapping.py

```diff
@@ -0,0 +1,127 @@
+                ),
+            ],
+        ),
+        ClickhouseQuery(
+            SnubaQuery(
+                body={},
+                data_source=TableSource("my_table", ColumnSet([])),
+                selected_columns=[
+                    Column("alias", "table", "column"),
+                    FunctionCall(
+                        "alias2",
+                        "f1",
+                        (Column(None, None, "column2"), Column(None, None, "column3")),
+                    ),
+                    SubscriptableReference(
+                        None, Column(None, None, "tags"), Literal(None, "myTag")
+                    ),
+                ],
+            )
+        ),
+        TranslationMappers(),
+    ),
+    (
+        "some basic rules",
+        SnubaQuery(
+            body={},
+            data_source=TableSource("my_table", ColumnSet([])),
+            selected_columns=[
+                Column("alias", "table", "column"),
+                FunctionCall(
+                    "alias2",
+                    "f1",
+                    (Column(None, None, "column2"), Column(None, None, "column3")),
+                ),
+                SubscriptableReference(
+                    "tags[myTag]", Column(None, None, "tags"), Literal(None, "myTag")
+                ),
+            ],
+        ),
+        ClickhouseQuery(
+            SnubaQuery(
+                body={},
+                data_source=TableSource("my_table", ColumnSet([])),
+                selected_columns=[
+                    Column("alias", "table", "column"),
+                    FunctionCall(
+                        "alias2",
+                        "f1",
+                        (
+                            Column(None, None, "not_column2"),
+                            Column(None, None, "column3"),
+                        ),
+                    ),
+                    FunctionCall(
+                        "tags[myTag]",
+                        "arrayElement",
+                        (
+                            Column(None, None, "tags.value"),
+                            FunctionCall(
+                                None,
+                                "indexOf",
+                                (
+                                    Column(None, None, "tags.key"),
+                                    Literal(None, "myTag"),
+                                ),
+                            ),
+                        ),
+                    ),
+                ],
+            )
+        ),
+        TranslationMappers(
+            columns=[SimpleColumnMapper(None, "column2", None, "not_column2")],
+            subscriptables=[TagMapper(None, "tags", None, "tags")],
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize("name, query, expected, mappers", test_cases)
+def test_translation(
+    name: str, query: SnubaQuery, expected: ClickhouseQuery, mappers: TranslationMappers
```

Comment:
nit: I think it's easier to read the test definitions if the mappers are defined before the input/output values.